minifier.js

```diff
@@ -1,116 +1,88 @@
   override: propOr('override', false, options),
-  exceptionUrls: toArrayIfNotAlreadyAnArray(propOr('exceptionUrls', propOr('exception_url', false, options), options)),
-  htmlMinifier: Object.assign(
-    {
-      removeComments: true,
-      collapseWhitespace: true,
-      collapseBooleanAttributes: true,
-      removeAttributeQuotes: true,
-      removeEmptyAttributes: true,
-    },
-    propOr('htmlMinifier', {}, options),
-  ),
+  // Create an array of exception urls
+  exceptionUrls: toArrayIfNotAlreadyAnArray(propOr('exceptionUrls', propOr('exception_url', [], options), options)),
+  // Merge user-defined htmlMinifier options with defaults
+  htmlMinifier: {
+    removeComments: true,
+    collapseWhitespace: true,
+    collapseBooleanAttributes: true,
+    removeAttributeQuotes: true,
+    removeEmptyAttributes: true,
+    ...propOr('htmlMinifier', {}, options),
+  },
 })
 
-const matchesException = (req, res) => {
-  return (exception) => {
-    if (exception instanceof RegExp) {
-      return exception.test(req.url)
-    }
-
-    if (typeof exception === 'function') {
-      return exception(req, res) || false
-    }
-
-    if (typeof exception === 'string') {
-      return req.url.match(exception) !== null
-    }
-
-    return false
+// Determine if the current request url matches any exceptions
+const matchesException = (req, res) => (exception) => {
+  if (exception instanceof RegExp) {
+    return exception.test(req.url)
   }
-}
 
-const sendMinified = (res, next, htmlMinifierOptions, callback) => {
-  if (typeof callback === 'undefined') {
-    // No callback specified, just minify and send to client.
-    return (err, html) => {
-      if (err) {
-        return next(err)
-      }
+  if (typeof exception === 'function') {
+    return exception(req, res) || false
+  }
 
-      minify(html, htmlMinifierOptions)
-        .then((html) => {
-          res.send(html)
-        })
-        .catch((err) => {
-          res.status(500).send(err)
-        })
-    }
+  if (typeof exception === 'string') {
+    return req.url.includes(exception)
   }
 
-  // Custom callback specified by user, use that one
-  return (err, html) => {
-    if (!html) {
-      callback(err, html)
-      return
-    }
+  return false
+}
 
-    minify(html, htmlMinifierOptions)
-      .then((html) => {
-        callback(err, html)
-      })
-      .catch((err) => {
-        callback(err)
-      })
+// Function to minify and send html response or call user-provided callback
+const sendMinified = (res, next, htmlMinifierOptions, callback) => (err, html) => {
+  // If an error occurred or html is not provided, call the callback immediately
+  if (err || !html) {
+    callback?.(err, html)
+    return
   }
+
+  // Minify the html and send it to the client or call the callback
+  minify(html, htmlMinifierOptions)
+    .then((html) => (callback ? callback(err, html) : res.send(html)))
+    .catch((err) => (callback ? callback(err) : res.status(500).send(err)))
```

Comment:
This looks way less readable compared to the previous version which first dealt with having no callback specified and afterwards it was free to use the callback. Instead of having a single check to see if callback is available now we have 3 on line 56, 62 and 63. Also the checks are insufficient: if I set callback to have a value of `15` it would raise an error.
I would suggest something like this:
```js
const callbackSpecified = typeof callback === 'function'

if (err || !html) {
  if (callbackSpecified) {
    callback(err, html)
  } else {
    next(err)
  }
}

minify(html, htmlMinifierOptions)
  .then(html => {
    if (callbackSpecified) {
      callback(err, html)
    } else {
      res.send(html)
    }
  })
  .catch(err => {
    if (callbackSpecified) {
      callback(err)
    } else {
      res.status(500).send(err)
    }
  })
```
This would make the code a bit more easier to upgrade to async/await later on and typescript will also be happy.

I would also suggest not using ternary operators as a 1-to-1 drop in replacement for if statements when not on the right side of an equasion as they can be a bit tricky to read.